ref-tests/src/utils.rs

```diff
@@ -25,7 +25,7 @@ pub async fn create_identity() -> Result<BasicIdentity, String> {
     ))
 }
 
-pub async fn create_agent() -> Result<Agent, String> {
+pub async fn create_agent(identity: BasicIdentity) -> Result<Agent, String> {
```

Comment:
Changed this here, now requring `create_identity()` be called separately instead of implicitly in `create_agent()`
This is to allow getting the Principal for that identity (which otherwise would have been private.)
Open to suggestion if this should be done a different way